### PR TITLE
Scene.draw etc.

### PR DESCRIPTION
**Pull request by Licini**
@tomvanmele Changes you requested:

- Moved registration to `get_scene_object_cls` so it will always be called.
- Renamed `redraw` to `draw`, and kept it as alias.
- Added pluggables `predraw` and `postdraw`.
- Also raise TypeError if item is not an `Data`, as alternative solution to #1274 .

### CODE REVIEW

File: src/compas/scene/scene.py [hunk 1]
```diff
@@ -302,8 +306,12 @@ def redraw(self):
         if drawn_objects:
             redraw()
 
+        postdraw()
+
         return drawn_objects
 
+    redraw = draw
```
Comment (Licini) [inline src/compas/scene/scene.py]: But to be honest, I would even vote for not keep this alias, to distinct it from the redraw pluggable

**Comment by tomvanmele** [comment]
do we even need the redraw pluggable? rhino's redraw can be triggered by the postdraw hook...

File: src/compas/scene/scene.py (hunk 1 above)
Comment (gonzalocasas) [reply src/compas/scene/scene.py]: Agree

File: src/compas/scene/context.py [hunk 2]
```diff
@@ -26,6 +26,22 @@ def redraw():
 redraw.__pluggable__ = True
 
 
+@pluggable(category="drawing-utils")
+def predraw():
```
Comment (gonzalocasas) [inline src/compas/scene/context.py]: Docstrings to explain what these do?

File: src/compas/scene/context.py [hunk 3]
```diff
@@ -26,6 +26,22 @@ def redraw():
...
+    pass
+
+
+predraw.__pluggable__ = True
+
+
+@pluggable(category="drawing-utils")
+def postdraw():
```
Comment (gonzalocasas) [inline src/compas/scene/context.py]: If these are hooks, I would call them `before_draw` and `after_draw`

**Review by gonzalocasas — APPROVED** [review]
LGTM

File: src/compas/scene/context.py (hunk 2 above)
Comment (tomvanmele) [reply src/compas/scene/context.py]: @Licini and please add them to `__init__` and to the api docs.

**Comment by Licini** [comment]
@tomvanmele All adjusted, take a final look?

**Review by tomvanmele — APPROVED** [review]

**Pull request merged**